configs/components/rubygem-ffi.rb

```diff
@@ -28,4 +28,44 @@
       "#{settings[:gem_install]} ffi-#{pkg.get_version}-#{platform.architecture}-mingw32.gem"
     end
   end
+
+  if platform.is_solaris? && platform.is_cross_compiled?
+    base_ruby = case platform.os_version
+                when "10"
+                  "/opt/csw/lib/ruby/2.0.0"
+                when "11"
+                  "/opt/pl-build-tools/lib/ruby/2.1.0"
+                end
+
+    case platform.os_version
+    when "10"
+      pkg.environment "PATH", "/opt/csw/bin:$$PATH"
+      pkg.install_file "/opt/csw/lib/libffi.so", "#{settings[:libdir]}/libffi.so.6"
+    when "11"
+      pkg.environment "PATH", "/opt/pl-build-tools/bin:$$PATH"
+      pkg.environment "CPATH", "/opt/csw/lib/libffi-3.2.1/include"
+      pkg.environment "MAKE", platform[:make]
+      pkg.install_file "#{settings[:tools_root]}/#{settings[:platform_triple]}/sysroot/usr/lib/libffi.so.5.0.10", "#{settings[:libdir]}/libffi.so"
+
+      # monkey-patch rubygems to think we're running on the destination ruby and architecture
+      ruby_api_version = settings[:ruby_version].gsub(/\.\d*$/, '.0')
```

Comment:
ruby_api_version should be the earliest compatible minor version (i.e. if we have ruby 2.4.9, the api version is 2.4.0)